server/log-collector/pkg/services/logcollector/server.go

```diff
@@ -754,10 +784,26 @@ func (s *Server) streamPodLogs(ctx context.Context,
 		}
 	}
 
-	// if error is EOF, the pod is done streaming logs (deleted/completed/failed)
+	// if error is EOF, it either means the pod is done streaming logs, or the logs in the apiserver were rotated,
+	// in such case we need to get a new stream
 	if err == io.EOF {
-		s.Logger.DebugWithCtx(ctx, "Pod logs are done streaming", "runUID", runUID)
-		return false, nil
+
+		pod, err := s.kubeClientSet.CoreV1().Pods(s.namespace).Get(ctx, podName, metav1.GetOptions{})
+		if err != nil {
+			return false, errors.Wrap(err, "Failed to get pod")
```

Comment:
if the pod was evicted we should keep the old behavior 
```suggestion
			return false, nil
```

Reply:
I can check whether the error is `Pod not found`, and if so return nil so everything stops properly.
Else, It can mean that some real other error occurred, and returning an error makes the whole process to be restarted by the monitoring loop.

WDYT?

Reply:
u can check the eviction flow easily
see https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#create-eviction-pod-v1-core